syntax = "proto3";

service EC2Operations {
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    rpc StoreData (StoreRequest) returns (StoreReply) {}
    rpc AppendData (AppendRequest) returns (AppendReply) {}
    rpc DeleteFile (DeleteRequest) returns (DeleteReply) {}
}

message StoreRequest {
    string data = 1;
}

message StoreReply {
    string s3uri = 1;
}

message AppendRequest {
    string data = 1;
}

message AppendReply {}

message DeleteRequest {
    string s3uri = 1;
}

message DeleteReply {}

message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}